mypy/semanal.py

```diff
@@ -277,6 +273,10 @@ def visit_file(self, file_node: MypyFile, fnam: str, options: Options,
         self.is_typeshed_stub_file = self.errors.is_typeshed_file(file_node.path)
         self.globals = file_node.names
         self.patches = patches
+        self.named_tuple_analyzer = NamedTupleAnalyzer(options, self)
```

Comment:
There is a bit strange asymmetry between `NamedTupleAnanlyzer` and `TypedDictAnalyzer`, because only the latter needs `self.msg`. It is needed by `check_for_explicit_any`, and it is a bit surprising that it doesn't appear in `NamedTupleAnalyzer`. I think it would be good or explain this asymmetry.